cmd/kueuectl/app/list/list_pods.go

```diff
@@ -0,0 +1,249 @@
+		Short:                 "List Pods belong to a Job Kind",
+		Long:                  podLong,
+		Example:               podExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cobra.CheckErr(o.Complete(clientGetter))
+			cobra.CheckErr(o.Run(cmd.Context()))
+		},
+	}
+
+	o.PrintFlags.AddFlags(cmd)
+
+	addAllNamespacesFlagVar(cmd, &o.AllNamespaces)
+	addFieldSelectorFlagVar(cmd, &o.FieldSelector)
+	addLabelSelectorFlagVar(cmd, &o.LabelSelector)
+	addForObjectFilterFlagVar(cmd, &o.UserSpecifiedForObject)
+
+	return cmd
+}
+
+// Complete takes the command arguments and infers any remaining options.
+func (o *PodOptions) Complete(clientGetter util.ClientGetter) error {
+	var err error
+
+	o.Namespace, _, err = clientGetter.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return err
+	}
+
+	o.Clientset, err = clientGetter.K8sClientSet()
+	if err != nil {
+		return err
+	}
+
+	if !o.PrintFlags.OutputFlagSpecified() {
+		o.PrintObj = printPodTable
+	} else {
+		printer, err := o.PrintFlags.ToPrinter()
+		if err != nil {
+			return err
+		}
+		o.PrintObj = printer.PrintObj
+	}
+
+	err = o.parseForObject()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Run prints the pods for a specific Job
+func (o *PodOptions) Run(ctx context.Context) error {
+	err := o.listPods(ctx)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *PodOptions) parseForObject() error {
+	if o.UserSpecifiedForObject == "" {
+		return nil
+	}
+
+	parts := strings.Split(o.UserSpecifiedForObject, "/")
+	if len(parts) > 2 {
+		return errors.New(fmt.Sprintf("invalid value '%s' used in --for flag; value must be in the format [TYPE[.API-GROUP]/]NAME", o.UserSpecifiedForObject))
+	}
+
+	if len(parts) == 1 {
+		o.ForObject.Name = parts[0]
+		return nil
+	}
+
+	o.ForObject.Name = parts[1]
+	o.ForObject.Kind, o.ForObject.APIGroup, _ = strings.Cut(parts[0], ".")
+
+	return nil
+}
+
+// listPods lists the pods based on the given --for object
+func (o *PodOptions) listPods(ctx context.Context) error {
+	continueToken := ""
+
+	for {
+		podList, err := o.Clientset.CoreV1().Pods(o.Namespace).List(ctx, metav1.ListOptions{
+			LabelSelector: o.LabelSelector,
+			FieldSelector: o.FieldSelector,
+			Continue:      continueToken,
+		})
+		if err != nil {
+			return err
+		}
+
+		if len(podList.Items) == 0 {
+			return nil
+		}
+
+		filteredPods := make([]corev1.Pod, 0, len(podList.Items))
+		for i := range podList.Items {
+			for _, ownerRef := range podList.Items[i].OwnerReferences {
+				gv, _ := schema.ParseGroupVersion(ownerRef.APIVersion)
+
+				if strings.EqualFold(o.ForObject.Kind, ownerRef.Kind) && strings.EqualFold(o.ForObject.Name, ownerRef.Name) {
+					if o.ForObject.APIGroup == "" || strings.EqualFold(o.ForObject.APIGroup, gv.Group) {
+						filteredPods = append(filteredPods, podList.Items[i])
+						break
+					}
+				}
+			}
+		}
+
+		// replace the podList items with the new filtered pods
+		podList.Items = filteredPods
```

Comment:
Can we move it to separate function?

Reply:
Moved in [b3eab73](https://github.com/kubernetes-sigs/kueue/pull/2280/commits/b3eab73f436a055029766f3cf84ad330b940825d)